src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js

```diff
@@ -167,6 +257,106 @@
                     $state.go('^');
                 });
             }]
+        })
+        .state('programming-exercise-for-course.delete', {
+            parent: 'programming-exercise-for-course',
+            url: '/{id}/delete',
+            data: {
+                authorities: ['ROLE_ADMIN', 'ROLE_TA']
```

Comment:
delete an exercise is something that only admins should be able to do, please remove the ROLE_TA here